code/modules/mob/living/simple_animal/hostile/pulse_demon/pulsedemon.dm

```diff
@@ -118,28 +119,57 @@
 	if(statpanel("Status"))
 		stat(null, text("Charge stored: [charge]W"))
 		stat(null, text("Max charge stored: [maxcharge]W"))
-
-/mob/living/simple_animal/hostile/pulse_demon/Life()
-	// Add the regen rate unless it puts us over max health, then just cap it off
+		
+/mob/living/simple_animal/hostile/pulse_demon/proc/power_lost()
+	health -= health_drain_rate
+	if(!powerloss_alerted)
+		to_chat(src, "You have lost power!")
+		powerloss_alerted = 1
+		//TODO add a sound
+		
+/mob/living/simple_animal/hostile/pulse_demon/proc/update_glow()
+	var/range = 1.5 
+	if(charge <= 50000) //why isn't there a log function, i guess i could have used len(str()) but that's fucking stupid
```

Comment:
why not a switching statement?
also, there is a log function, https://www.byond.com/docs/ref/#/proc/log